modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java

```diff
@@ -634,12 +645,10 @@ public ObjectDefinition putObjectDefinition(
 			_objectLayoutLocalService.deleteObjectLayouts(objectDefinitionId);
 		}
 
-		ObjectRelationship[] objectRelationships =
-			objectDefinition.getObjectRelationships();
-
 		Set<String> accountEntryRestrictedObjectRelationshipsNames =
 			_getAccountEntryRestrictedObjectRelationshipsNames(
-				serviceBuilderObjectDefinition, objectRelationships);
+				serviceBuilderObjectDefinition,
+				objectRelationships.toArray(new ObjectRelationship[0]));
```

Comment:
Isolate this in a new variable: `ObjectRelationship[] objectRelationships = objectRelationships.toArray(new ObjectRelationship[0])` for you to be able to use in both places

Reply:
Since objectRelationships already exists, you can create selectedObjectRelationships